src/main/java/com/websafe/ccmsbe/service/PrivacyRegulationService.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public PrivacyRegulation createPrivacyRegulation(PrivacyRegulation privacyRegulation) {
+        try {
+            if (privacyRegulation == null) {
+                throw new IllegalArgumentException("Privacy regulation cannot be null");
+            }
+
+            return privacyRegulationRepository.save(privacyRegulation);
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new PrivacyRegulationCreationException("An error occurred while creating the privacy regulation");
+        }
+    }
+
+    public List<PrivacyRegulation> getAllPrivacyRegulations() {
+        try {
+            return privacyRegulationRepository.findAll();
+        } catch (Exception e) {
+            throw new PrivacyRegulationRetrievalException("An error occurred while retrieving all privacy regulations");
+        }
+    }
+
+    @Transactional
+    public String addPrivacyRegulationToWebsite(Long websiteId, Long regulationId) {
+        try {
+            Website website = websiteRepository.findById(websiteId).orElse(null);
+
+            PrivacyRegulation privacyRegulation = privacyRegulationRepository.findById(regulationId).orElse(null);
+
+            if (website == null || privacyRegulation == null) {
+                throw new PrivacyRegulationNotFoundException("Invalid website ID or regulation ID");
+            }
+
+            if (website.getPrivacyRegulations().contains(privacyRegulation)) {
+                throw new PrivacyRegulationNotFoundException("Regulation already added to the website");
+            }
+
+            website.getPrivacyRegulations().add(privacyRegulation);
+
+            websiteRepository.save(website);
+
+            return "Successfully added";
+
+        } catch (PrivacyRegulationNotFoundException e) {
+            return e.getMessage();
+        } catch (Exception e) {
+            return "An error occurred while adding the privacy regulation";
+        }
+    }
+
+    @Transactional
+    public String deletePrivacyRegulationFromWebsite(Long websiteId, Long regulationId) {
+        try {
+            Website website = websiteRepository.findById(websiteId).orElseThrow(
+                    () -> new WebsiteNotFoundException("Website not found with ID: " + websiteId)
+            );
+
+            PrivacyRegulation privacyRegulation = privacyRegulationRepository.findById(regulationId).orElseThrow(
+                    () -> new PrivacyRegulationNotFoundException("Privacy regulation not found with ID: " + regulationId)
+            );
+
+            List<PrivacyRegulation> privacyRegulations = website.getPrivacyRegulations();
+
+            boolean removed = privacyRegulations.removeIf(privacyRegulation1 -> privacyRegulation1.getRegulationId().equals(regulationId));
+
+            if (removed) {
+                website.setPrivacyRegulations(privacyRegulations);
+                websiteRepository.save(website);
+
```

Comment:
No need to keep a blank line before the return statement. Check other places as well.